src/handlers/pointer_handler.rs

```diff
@@ -0,0 +1,91 @@
+use bevy::{
+    input::{ButtonState, mouse::MouseButtonInput},
+    prelude::MouseButton,
+    window::{CursorEntered, CursorLeft, CursorMoved, Window, WindowEvent},
+};
+use smithay_client_toolkit::{
+    delegate_pointer, reexports::client::Proxy, seat::pointer::PointerHandler,
+};
+
+use crate::{smithay_windows::SmithayWindows, state::SmithayRunnerState};
+
+fn convert_to_mouse_button(button: u32) -> MouseButton {
+    match button {
+        272 => MouseButton::Left,
+        273 => MouseButton::Right,
+        274 => MouseButton::Middle,
+        277 => MouseButton::Forward,
+        278 => MouseButton::Back,
+        other => MouseButton::Other(other as u16),
+    }
+}
+
+impl PointerHandler for SmithayRunnerState {
+    fn pointer_frame(
+        &mut self,
+        _: &smithay_client_toolkit::reexports::client::Connection,
+        _: &smithay_client_toolkit::reexports::client::QueueHandle<Self>,
+        _: &smithay_client_toolkit::reexports::client::protocol::wl_pointer::WlPointer,
+        events: &[smithay_client_toolkit::seat::pointer::PointerEvent],
+    ) {
+        for event in events {
+            let smithay_windows = self.world().non_send_resource::<SmithayWindows>();
+            let window_id = event.surface.id();
+            let entity = smithay_windows.smithay_to_entity.get(&window_id);
+            if entity.is_none() {
+                return;
+            }
+            let entity = *entity.unwrap();
+            let window = self.world().get::<Window>(entity).unwrap().clone();
+            let mut position = bevy::math::Vec2 {
+                x: event.position.0 as f32,
+                y: event.position.1 as f32,
+            };
+            let delta = window
+                .physical_cursor_position()
+                .map(|old_position| (position - old_position) / window.scale_factor());
+            let pointer_event: WindowEvent = match event.kind {
+                smithay_client_toolkit::seat::pointer::PointerEventKind::Enter { .. } => {
```

Comment:
In all these enum cases you have considered args as `{ .. }`,  do we not need to handle any?